accesstoken2/src/app.ts

```diff
@@ -72,17 +54,28 @@ export class AccessTokenLambda implements LambdaInterface {
                 statusCode: 200,
                 body: JSON.stringify(accessTokenResponse),
             };
-        } catch (err: any) { //Todo dont want any
+        } catch (err: unknown) { //Todo dont want any
+            for (const appError of ApplicationErrors) {
+                if (err instanceof appError) {
+                    logError(err);
+                    return getError(err);
+                }
+            }
+            logError(err as unknown as BaseError);
+            return Promise.reject({statusCode: 500, message: "Unknown Error Occurred"});
+        }
+        function logError(err: CustomErrors) {
             logger.error({
                 statusCode: err.statusCode ?? 500,
-                message: err.message,
+                message: err?.message,
                 err: err
             });
-
+        }
+        function getError(err: CustomErrors): APIGatewayProxyResult | PromiseLike<APIGatewayProxyResult> {
```

Comment:
we should avoid functions that have two return types that could be a promise sometimes, and sometimes are not a promise

Reply:
Yeah, seems to be fine with `APIGatewayProxyResult`